OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -0,0 +1,757 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_VULKAN_LAYER_COMMAND_BUFFER_MANAGER_H_
+#define ORBIT_VULKAN_LAYER_COMMAND_BUFFER_MANAGER_H_
+
+#include <stack>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Profiling.h"
+#include "VulkanLayerProducer.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "absl/synchronization/mutex.h"
+#include "vulkan/vulkan.h"
+
+namespace orbit_vulkan_layer {
+
+namespace internal {
+
+// Stores meta information about a submit (VkQueueSubmit), e.g. to allow to identify the matching
+// Gpu tracepoint.
+struct SubmissionMetaInformation {
+  uint64_t pre_submission_cpu_timestamp;
+  uint64_t post_submission_cpu_timestamp;
+  int32_t thread_id;
+};
+
+// A persistent version of a command buffer that was submitted and its begin/end slot in the
+// `TimerQueryPoool`. Note that the begin is optional, as it might not be part of the capture.
```

Comment:
So, how do we visualize timeslices that don't have the begin? Do we throw them away?

Reply:
We use the "begin of capture" as begin for now. There has been a bug around about this in general already a long time ago. We at one point need a way to visualize this clearly.